pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -49,6 +49,7 @@ while [ "$#" -gt 0 ]; do
         ;;
       switch|boot|test|build|edit|dry-build|dry-run|dry-activate|build-vm|build-vm-with-bootloader)
         if [ "$i" = dry-run ]; then i=dry-build; fi
+        if [ -n "$action" ]; then showSyntax; fi
```

Comment:
`action` is zero-initialised in line 24 and holds exactly one argument.

In this `getopt` like loop, positional arguments (`switch`, `boot`, etc.) are parsed together with flags (`--help`, etc.) and the code only saves the last argument given.

So consider the first time hitting this case:  `action` is still empty and will be set to the intended action below.
If users errorneously specify a second argument (which is invalid according to nixos-rebuild(1) SYNOPSIS), this case is hit again, but `action` is not empty and the old action gets overwritten.

One could hoist mandatory action parsing before optional flags parsing but the way `nixos-rebuild` works, the check would practically be the same.

Reply:
This explanation would be better made in a comment in the code itself, rather than a github comment that no one would ever see again.

Reply:
Sure, I'll add a comment.

(It simply didn't occur that me that this fix would be non-obvious to anyone.)